File: python/src/buble_ai/types/files.py
from __future__ import annotations

from typing import Literal, TypedDict

UploadFileType = Literal["image", "video", "audio"]


class UploadedFile(TypedDict):
    object: str
    provider: str
    url: str
    key: str
    file_type: UploadFileType
    content_type: str
    size: int
    filename: str


class UploadedFileEnvelope(TypedDict):
    data: UploadedFile

File: npm/src/types/files.ts
import type { ApiEnvelope, MediaType } from './common.js';

export type UploadFileType = MediaType;

export type UploadFileInput =
  | string
  | Uint8Array
  | ArrayBuffer
  | Blob
  | NodeJS.ReadableStream;

export type UploadFileOptions = {
  file_type?: UploadFileType;
  fileType?: UploadFileType;
  model?: string;
  mode?: string;
  filename?: string;
  contentType?: string;
  signal?: AbortSignal;
  timeout?: number;
};

export type UploadedFile = {
  object: 'file';
  provider: string;
  url: string;
  key: string;
  file_type: UploadFileType;
  content_type: string;
  size: number;
  filename: string;
};

export type UploadedFileEnvelope = ApiEnvelope<UploadedFile>;
